src/modules/fundingManager/oracle/FM_PC_Oracle_Redeeming_v1.sol

```
@@ -0,0 +1,1040 @@
+        external
+        virtual
+        override(BondingCurveBase_v1, IBondingCurveBase_v1)
+        onlyOrchestratorAdmin
+    {
+        _setBuyFee(fee_);
+    }
+
+    /// @inheritdoc IFM_PC_Oracle_Redeeming_v1
+    function setProjectTreasury(address projectTreasury_)
+        external
+        virtual
+        onlyOrchestratorAdmin
+    {
+        _setProjectTreasury(projectTreasury_);
+    }
+
+    /// @inheritdoc IFM_PC_Oracle_Redeeming_v1
+    function setOracleAddress(address oracle_)
+        external
+        virtual
+        onlyOrchestratorAdmin
+    {
+        _setOracleAddress(oracle_);
+    }
+
+    /// @inheritdoc IFM_PC_Oracle_Redeeming_v1
+    function getBuyFee() public view virtual returns (uint buyFee_) {
+        return buyFee;
+    }
+
+    /// @inheritdoc IFM_PC_Oracle_Redeeming_v1
+    function getMaxProjectBuyFee()
+        public
+        view
+        virtual
+        returns (uint maxProjectBuyFee_)
+    {
+        return _maxProjectBuyFee;
+    }
+
+    /// @inheritdoc IFM_PC_Oracle_Redeeming_v1
+    function getMaxProjectSellFee()
+        public
+        view
+        virtual
+        returns (uint maxProjectSellFee_)
+    {
+        return _maxProjectSellFee;
+    }
+
+    /// @inheritdoc IFM_PC_Oracle_Redeeming_v1
+    function setIsDirectOperationsOnly(bool isDirectOperationsOnly_)
+        public
+        virtual
+        onlyOrchestratorAdmin
+    {
+        _setIsDirectOperationsOnly(isDirectOperationsOnly_);
+    }
+
+    /// @inheritdoc IFM_PC_Oracle_Redeeming_v1
+    function executeRedemptionQueue()
+        external
+        virtual
+        onlyModuleRole(QUEUE_EXECUTOR_ROLE)
+    {
+        (bool success,) = address(__Module_orchestrator.paymentProcessor()).call(
+            abi.encodeWithSignature(
+                "executePaymentQueue(address)", address(this)
+            )
+        );
+        if (!success) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_QueueExecutionFailed();
```

it is never a good idea to suppress error messages - you probably want to include the reason for the revert in the Error in some way

---

The idea here was that the only reason it would fail is because the contract being called doesn't have the correct function. However we can add the `bytes data` from the call to the custom error

---

Has been addressed